# TONG CODE

# Imports and Version Checking

In [18]:
import librosa
from librosa import display
import sklearn
import torch

# python version
import sys
print("Python version: " + sys.version)

# librosa version
# ???

# sklearn
print("Sklearn version: " + sklearn.__version__)

# torch
print("Torch version: " + torch.__version__)

Python version: 3.6.8 (default, Oct  7 2019, 12:59:55) 
[GCC 8.3.0]
Sklearn version: 0.21.3
Torch version: 1.0.1.post2


# Data Loading

In [28]:
genres = ['blues', 'classical', 'country', 'disco', 'hiphop', 'jazz', 'metal', 'pop', 'reggae', 'rock']

single_numbers = ['00000','00001','00002','00003','00004','00005','00006','00007','00008','00009']
double_numbers = list(["000" + str(n) for n in range(10,100)])
song_numbers = single_numbers + double_numbers

paths = []
for genre in genres:
    genre_paths = []
    for number in song_numbers:
        path = "genres/" + genre + "/" + genre + "." + number + ".au"
        genre_paths.append(path)
    paths.append(genre_paths)
    
# print(paths)
# print(paths[0])
# print(paths[0][0])

# Finding the min length

In [42]:
min_song_length = 670000
for genre_list in paths:
    for song in genre_list:
        y, _ = librosa.load(song)
        if y.shape[0] < min_song_length:
            min_song_length = y.shape[0]

print(min_song_length)

660000


# Generating MFCC on arbitrary song

In [41]:
y, sr = librosa.load(paths[0][7])

hop_length = 512
n_mfcc = 13
mfcc = librosa.feature.mfcc(y=y[:min_song_length], sr=sr, hop_length=hop_length, n_mfcc=n_mfcc)

print(mfcc.shape)

(13, 1290)


http://cs229.stanford.edu/proj2018/report/19.pdf for DCNNs

TODO:

- Develop MFCCs
- Work under assumption that we are flattening MFCCs for each song
- Develop architecture for DCNN under those assumptions (13 * 1290 X 1 input data point)
